agent/agent_pool.go

```diff
@@ -75,12 +79,21 @@ func (r *AgentPool) Start() error {
 
 	// Start a signalwatcher so we can monitor signals and handle shutdowns
 	signalwatcher.Watch(func(sig signalwatcher.Signal) {
+		r.signalLock.Lock()
+		defer r.signalLock.Unlock()
+
 		if sig == signalwatcher.QUIT {
 			logger.Debug("Received signal `%s`", sig.String())
 			worker.Stop(false)
 		} else if sig == signalwatcher.TERM || sig == signalwatcher.INT {
-			logger.Debug("Received signal `%s`", sig.String())
-			worker.Stop(true)
+			if r.interruptCount == 0 {
+				r.interruptCount++
+				logger.Debug("Received initial signal `%s`", sig.String())
```

Comment:
Should we do a `logger.Info` of "Received CTRL-C, send again to forcefully kill the agent" or some such?